File: 5-Association_Rule_Learning/1-Apriori/src/apriori.R
# Apriori Associated Rule Learning

# Path to the file
BinPath = paste(getwd(), 'MachineLearning-Portfolio/5-Association_Rule_Learning/1-Apriori/bin', sep='/')
FilePath = paste(BinPath, '/Market_Basket_Optimisation.csv', sep='')
setwd(BinPath)

# Importing Libraries
#install.packages('caTools', repos = "http://cran.us.r-project.org",dependencies = TRUE)
#install.packages('arules',repos = "http://cran.us.r-project.org", dependencies = TRUE)
#install.packages('arulesViz',repos = "http://cran.us.r-project.org", dependencies = TRUE)

library(caTools, help, pos = 2, lib.loc = NULL)
library(arules)
library(arulesViz)

# Import dataset
dataset = read.csv(FilePath, header = FALSE)
dataset = read.transactions(FilePath, sep=",", rm.duplicates = TRUE)
summary(dataset)

# Getting the top frequently bought items
png("Top100_R.png") 
itemFrequencyPlot(dataset, topN = 100)
dev.off() 

png("Top10_R.png")
itemFrequencyPlot(dataset, topN = 10)
dev.off() 

# Training Apriori on the dataset
rules = apriori(data = dataset,
                parameter = list(support = 0.004, confidence = 0.2))

# Visualising the results
jpeg("ParacoordList_R.jpeg")
plot(sort(rules, by = 'lift')[1:10], 
    method = "paracoord", 
    control = list(verbose = TRUE))
dev.off()

File: 5-Association_Rule_Learning/1-Apriori/src/apriori.py
# Apriori Associated Rule Learning

# Import the Libraries
import pandas as pd
import os

from apyori import apriori

# Import the data
BinPath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'bin'))
FilePath = BinPath + '\Market_Basket_Optimisation.csv';
dataset = pd.read_csv(FilePath, header = None)

# Moving the dataset to a list of transaction string
transactions = []
for i in range(0, 7501):
    transactions.append([str(dataset.values[i,j] for j in range(0, 20))])

# Training Apriori on the dataset
rules = apriori(transactions, 
                min_support = 0.003, 
                min_confidence = 0.2, 
                min_lift = 3, 
                min_length = 2)

# Visualising the results
results = list(rules)
print(results)